packages/toolkit/src/view/knowledge/components/CatalogFilesTab.tsx

```diff
@@ -0,0 +1,508 @@
+import React from 'react';
+import { Button, Icons, Separator, Skeleton, Tag } from "@instill-ai/design-system";
+import { KnowledgeBase, File } from "../../../lib/vdp-sdk/knowledge/types";
+import { DELETE_FILE_TIMEOUT } from "./undoDeleteTime";
+import DeleteFileNotification from "./Notifications/DeleteFileNotification";
+import { InstillStore, useInstillStore, useShallow } from "../../../lib";
+import { useListKnowledgeBaseFiles, useDeleteKnowledgeBaseFile } from "../../../lib/react-query-service/knowledge";
+import FileDetailsOverlay from "./FileDetailsOverlay";
+
+type CatalogFilesTabProps = {
+  knowledgeBase: KnowledgeBase;
+};
+
+type FileStatus = "NOTSTARTED" | "WAITING" | "CONVERTING" | "CHUNKING" | "EMBEDDING" | "COMPLETED" | "FAILED";
+
+type TagVariant = "lightNeutral" | "lightYellow" | "default" | "lightGreen" | "lightRed";
+
+const getStatusTag = (status: FileStatus): { variant: TagVariant; dotColor: string } => {
+  const statusMap: Record<FileStatus, { variant: TagVariant; dotColor: string }> = {
+    NOTSTARTED: { variant: "lightNeutral", dotColor: "bg-semantic-fg-secondary" },
+    WAITING: { variant: "lightYellow", dotColor: "bg-semantic-warning-default" },
+    CONVERTING: { variant: "default", dotColor: "bg-semantic-accent-default" },
+    CHUNKING: { variant: "default", dotColor: "bg-semantic-accent-default" },
+    EMBEDDING: { variant: "default", dotColor: "bg-semantic-accent-default" },
+    COMPLETED: { variant: "lightGreen", dotColor: "bg-semantic-success-default" },
+    FAILED: { variant: "lightRed", dotColor: "bg-semantic-error-default" },
+  };
+  return statusMap[status] || statusMap.NOTSTARTED;
+};
+
+export const CatalogFilesTab: React.FC<CatalogFilesTabProps> = ({ knowledgeBase }) => {
+  const [sortConfig, setSortConfig] = React.useState<{
+    key: keyof File | "";
+    direction: "ascending" | "descending" | "";
+  }>({
+    key: "",
+    direction: "",
+  });
+
+  const { accessToken, enabledQuery } = useInstillStore(
+    useShallow((store: InstillStore) => ({
+      accessToken: store.accessToken,
+      enabledQuery: store.enabledQuery,
+    }))
+  );
+
+  const { data: files, isLoading, refetch } = useListKnowledgeBaseFiles({
+    ownerId: knowledgeBase.ownerName,
+    knowledgeBaseId: knowledgeBase.kbId,
+    accessToken,
+    enabled: enabledQuery,
+  });
```

Comment:
Typically, we don't do the destruction of the react-query return since we might have multiple it within the same component, so it's better to keep it as singular variable and use it. Like

const knowledgeBaseFiles = ...

knowledgeBaseFiles.refetch()

This also guarantee better readability